recorder.go

```diff
@@ -349,21 +353,22 @@ func (r *Recorder) Flush() {
 	r.reportInFlight = false
 	if err != nil {
 		// Restore the records that did not get sent correctly
-		r.counters.droppedSpans += int64(r.buffer.addSpans(rawSpans))
-		r.counters.droppedSpans += droppedPending
+		atomic.AddInt64(&r.counters.droppedSpans, int64(r.buffer.addSpans(rawSpans))+droppedPending)
 		r.lock.Unlock()
 		return
 	}
 
 	// Reset the buffers
 	r.reportOldest = now
 	r.reportYoungest = now
-	// TODO: this ends up discarding counts coming in during the RPC
-	r.counters = counterSet{}
```

Comment:
This is definitely a bug, so great. I'm not convinced that the atomic int is needed (isn't it always read/written under the lock?) but happy to hear more.